.github/workflows/build.yml

```diff
@@ -32,25 +32,28 @@ jobs:
 
   # === OS Specific Job (runs on each OS) ===
   os_specific:
-    name: ${{ matrix.platform }}
+    name: ${{ matrix.sys.os }}
     timeout-minutes: 90
     strategy:
       matrix:
         go-version:
           - 1.20.x
-        platform:
-          - ubuntu-20.04
-          - macos-11
-          - windows-2019
+        sys:
+          - {os: ubuntu-20.04, shell: bash}
+          - {os: macos-11, shell: 'zsh {0}'}
+          - {os: windows-2019, shell: bash}
       fail-fast: false
-    runs-on: ${{ matrix.platform }}
+    defaults:
+      run:
+        shell: ${{ matrix.sys.shell }}
```

Comment:
Credit goes to https://github.com/orgs/community/discussions/27143#discussioncomment-3254801 for this solution.